src/line.rs

```diff
@@ -277,14 +233,24 @@ impl<'input, 'header, Endian> StateMachine<'input, 'header, Endian>
     /// is complete, and there are no more new rows in the line number matrix,
     /// then `Ok(None)` is returned. If there was an error parsing an opcode,
     /// then `Err(e)` is returned.
-    pub fn next_row(&mut self) -> parser::ParseResult<Option<LineNumberRow<'input, 'header, Endian>>> {
+    pub fn next_row(&mut self)
+                    -> parser::ParseResult<Option<&LineNumberRow<'input, 'header, Endian>>> {
+        if self.row.end_sequence {
+            self.row.reset();
+        } else {
+            self.row.discriminator = 0;
+            self.row.basic_block = false;
+            self.row.prologue_end = false;
+            self.row.epilogue_begin = false;
```

Comment:
A comment here about Steps 3-7 from 6.5.2.1 would be nice. Is it valid to always apply this after yielding a row, not just for special opcodes?